File: Grind 75 in Java/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> count = new HashMap<Character, Integer>();

        for (char c : s.toCharArray())
            count.put(c, count.getOrDefault(c, 0) + 1);

        int res = 0;
        for (int value : count.values()) {
            res += (value % 2 == 0) ? value : value - 1;
            if (res % 2 == 0 && value % 2 == 1)
                res++;
        }

        return res;
    }
}

File: leetcode in kotlin/0409-longest-palindrome.kt
class Solution {
    fun longestPalindrome(s: String): Int {
        val hm = HashMap<Char, Int>()
        var res = 0
        for(c in s)
            hm[c] = hm.getOrDefault(c, 0) + 1
        for((c,i) in hm){
            res += i/2 * 2
            if(res%2 == 0 && i%2 == 1) //if (only once) we have an odd char count
                res++
        }
        return res
    }
}
